modules/ipi-install-configuring-host-network-interfaces-in-the-install-config.yaml-file.adoc

```diff
@@ -0,0 +1,56 @@
+// This is included in the following assemblies:
+//
+// installing_bare_metal_ipi/ipi-install-installation-workflow.adoc
+
+[id="configuring-host-network-interfaces-in-the-install-config.yaml-file_{context}"]
+= (Optional) Configuring host network interfaces in the `install-config.yaml` file
+
+During installation, you can set the `networkConfig` configuration setting in the `install-config.yaml` file to configure host network interfaces using NMState on the `baremetal` network. To use the `networkConfig` configuration setting, you must pass an NMState key file as YAML. See link:https://nmstate.io/examples.html#interfaces-ethernet[NMState] for additional examples of the NMState syntax.
+
+.Example
+[source,yaml]
+----
+  hosts:
+        - name: openshift-master-0
+          role: master
+          bmc:
+            address: redfish+http://<out-of-band-ip>/redfish/v1/Systems/
+            username: <user>
+            password: <password>
+            disableCertificateVerification: null
+          bootMACAddress: <NIC1_mac_address>
+          bootMode: UEFI
+          rootDeviceHints:
+            deviceName: "/dev/sda"
+          networkConfig:
+            interfaces:
+            - name: <NIC1_name>
+              type: ethernet
+              state: up
+              ipv4:
+                address:
+                - ip: "<IP_address>"
+                  prefix-length: 24
+                enabled: true
+            dns-resolver:
+              config:
+                server:
+                - <DNS_IP_address>
+            routes:
+              config:
+              - destination: 0.0.0.0/0
+                next-hop-address: <IP_address>
+                next-hop-interface: <NIC1_name>
+----
+
+[TIP]
+====
+Consider testing out NMState configurations using the NMState command line interface before including them in the `install-config.yaml` file, because the installer will not check the NMState syntax of the YAML string. Errors in the YAML string syntax might result in a failure to apply the network configuration. Additionally, maintaining the validated YAML string syntax is useful when applying changes using Kubernetes NMState after deployment or when expanding the cluster.
```

Comment:
Can we include the command to do this?

nmstatectl gc [file containing the networkConfig YAML]